Source/Other/Extensions/StringRegexExtension.swift

```diff
@@ -0,0 +1,37 @@
+//
+//  StringRegexExtension.swift
+//  sequel-ace
+//
+//  Created by James on 22/12/2020.
+//  Copyright © 2020 Sequel-Ace. All rights reserved.
+//
+
+import Foundation
+
+
+extension String {
+
+    /// Returns the first string group matching the regex. A replacement for RegexLite's stringByMatching:(NSString *)regex capture:(NSInteger)capture
+    /// It's a bit slower than RegexLite but not called often and means we can exclude RegexLite from the tunnel assistant and get some logging into RegexLite.
+    /// - Parameters:
+    ///   - regex: The regular expression - must contain group captures. e.g. "^\\s*Enter passphrase for key \\'(.*)\\':\\s*$"
+    /// - Returns: The string matching the first group, or an empty string
+    func captureGroup(regex: String) -> String {
+        let regexValid = try? NSRegularExpression(pattern: regex, options: [])
+        if let match = regexValid?.firstMatch(in: self, options: [], range: NSRange(self.startIndex..<self.endIndex, in: self)) {
```

Comment:
I would put both if conditions into one and single `guard` :)

Reply:
dammit